src/lib/utils/log_manager.cpp

```diff
@@ -0,0 +1,26 @@
+#include "log_manager.hpp"
+
+#include "constant_mappings.hpp"
+#include "utils/settings/log_level_setting.hpp"
+
+namespace opossum {
+
+LogManager::LogManager(SettingsManager& settings_manager, LogLevel log_level) : _log_level(log_level) {
```

Comment:
I'd retrieve the settings_manager from the Hyrise object. We don't pass around pseudo singletons (e.g., the StorageManager) in other places, so we don't need to start here.

Reply:
The problem is that both Managers get initialized when Hyrise is initialized or reset. On Hyrise init, this causes infinite recursion, as Hyrise::get() keeps on creating new instances, on reset, we register the setting to the old manager that is reset, too.

Reply:
Got it. Maybe add a comment?